### PR TITLE
[orc] Bump to 1.9.0 & fix usage

### LINKED ISSUE #33015: [orc] update to 1.9.0
### Library name

orc

### New version number

1.9.0

### Other information that may be useful (release notes, etc...)

The current usage file of the port also includes `find_package` calls to orc's dependencies. These should be replaced by `find_dependeny()`-calls in orc's cmake config file.

### PR DESCRIPTION
**Pull request by xiaozhuai**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Close #33015

### CODE REVIEW

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
Tested usage successfully by `orc:x64-windows`:
````
    find_package(unofficial-orc CONFIG REQUIRED)
    target_link_libraries(main PRIVATE unofficial::orc::orc)
````

File: ports/orc/fix-cmake.patch [hunk 1]
```diff
@@ -0,0 +1,355 @@
+diff --git a/c++/src/CMakeLists.txt b/c++/src/CMakeLists.txt
```
Comment (JavierMatosD) [inline ports/orc/fix-cmake.patch:1]: @xiaozhuai, This patch seems OK, but it is pretty long. Could you disable the unwanted blocks with `if(0)` rather than deleting it?

Comment (xiaozhuai) [reply ports/orc/fix-cmake.patch:1]: Done~

File: ports/orc/portfile.cmake [hunk 2]
```diff
@@ -1,12 +1,13 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```
Comment (Thomas1664) [inline ports/orc/portfile.cmake:1]: Why did you add `ONLY_STATIC_LIBRARY`?

Comment (xiaozhuai) [reply ports/orc/portfile.cmake:1]: @Thomas1664 
Because it is.
See https://github.com/apache/orc/blob/9d434392e46252bf6ad2a6414527d2e619f858e6/c%2B%2B/src/CMakeLists.txt#L209
